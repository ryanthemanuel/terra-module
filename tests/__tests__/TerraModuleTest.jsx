import React from 'react';
import { shallow } from 'enzyme';
import Module from '../../src/TerraModule';

const defaultVariety = <Module />;
const primaryVariety = <Module name="primary" variant="terra-Module--primary" />;


// Snapshot Tests
it('should render a default component', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
});

it('should render a primary component', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper).toMatchSnapshot();
});


// Prop Tests
it('should have the class terra-Module--default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper.prop('className')).toContain('terra-Module terra-Module--default');
});

it('should have the class terra-Module--primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper.prop('className')).toContain('terra-Module terra-Module--primary');
});


// Event Tests
it('should toggle the class u-selected on default', () => {
  const wrapper = shallow(defaultVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Module--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Module--default').simulate('click');
  expect(wrapper).toMatchSnapshot();
});

it('should toggle the class u-selected on primary', () => {
  const wrapper = shallow(primaryVariety);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Module--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
  wrapper.find('.terra-Module--primary').simulate('click');
  expect(wrapper).toMatchSnapshot();
});
