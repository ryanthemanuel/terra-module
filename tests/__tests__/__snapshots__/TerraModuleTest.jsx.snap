exports[`test should render a default component 1`] = `
<button
  className="terra-Module terra-Module--default"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should render a primary component 1`] = `
<button
  className="terra-Module terra-Module--primary"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;

exports[`test should toggle the class u-selected on default 1`] = `
<button
  className="terra-Module terra-Module--default"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should toggle the class u-selected on default 2`] = `
<button
  className="terra-Module terra-Module--default u-selected"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should toggle the class u-selected on default 3`] = `
<button
  className="terra-Module terra-Module--default"
  onClick={[Function]}>
  Terra, 
  default
</button>
`;

exports[`test should toggle the class u-selected on primary 1`] = `
<button
  className="terra-Module terra-Module--primary"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;

exports[`test should toggle the class u-selected on primary 2`] = `
<button
  className="terra-Module terra-Module--primary u-selected"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;

exports[`test should toggle the class u-selected on primary 3`] = `
<button
  className="terra-Module terra-Module--primary"
  onClick={[Function]}>
  Terra, 
  primary
</button>
`;
